### DISCUSSION TITLE
Placing the colorchecker at any illuminant

### DISCUSSION THREAD

**Discussion by xeggbee** [colour-science/colour#1089] [Q&A] [answered]
I`d like to confirm if is possible to place the colorchecker at ANY illuminant.

Here is the experiment:

### Part 1: from babel colorchecker to srgb with generates d65 illuminant

1- From Babel ColorChecker, took the dark skin spectral distibrution, which is on illumiannd D50

`sdred = SDS_COLOURCHECKERS['BabelColor Average']['dark skin']`
`sd = SpectralDistribution(sdred)`

now, the darkskin spectral lieves on sd;

2- Generating the xy from D65: (if plot the SDS_ILLUMINANTS['D65'] its quite similar!)

`CCT_D65 = 6500 * 1.4388 / 1.4380`
`xy = CCT_to_xy_CIE_D(CCT_D65)`
`sd_illuminant = colour.sd_CIE_illuminant_D_series(xy)`

now, the generate d65 illuminante lieves on sd_illuminant;

3-Converting the darkskin spectral distribution from babel which is d50 to generated d65 illuminant:

`illuminant = SDS_ILLUMINANTS['D65']`
`convert(sd, 'Spectral Distribution', 'sRGB',sd_to_XYZ={'illuminant': sd_illuminant}, verbose={'mode': 'Long'})`

As SRGB value, we have the following:

`[ 0.45226021  0.32042024  0.2661405 ] `

### Part 2: Using a spectrophotometer capture the darkskin from colorchecker on D65 Illumiant [400 to 700nm, interval 10]

1-read the file and convert it srgb

`spectral = spectral.reshape(31)`

# SHAPE 
`w = SpectralShape(400, 700, 10).range()`
`sdred = SpectralDistribution(spectral,w, name="Red")`
`plot_single_sd(sdred)`

# CONVERT 
`sd = SpectralDistribution(sdred)`
`cmfs = MSDS_CMFS['CIE 1931 2 Degree Standard Observer']`
`illuminant = SDS_ILLUMINANTS['D65']`

# Calculating the sample spectral distribution *CIE XYZ* tristimulus values.
`XYZ = sd_to_XYZ(sd, cmfs, illuminant)`
`RGB = XYZ_to_sRGB(XYZ /10000)` ## this spectrophotomer requires to divide by 10.000

As SRGB value, we have the following:

`0.42998547  0.30259963  0.25227298`

### Part 3: delta E between Part 1 and Part 2

1-from SRGB to Lab

`lab1 = sRGB_to_XYZ([0.45226056,0.32042014,0.2661407 ])`
`lab1 = XYZ_to_Lab(lab1)`
`LAB1 -> [ 37.97355444  12.10513798  13.69724455]`

`lab2 = sRGB_to_XYZ([0.42998547,  0.30259963,  0.25227298])`
`lab2 = XYZ_to_Lab(lab2)`
`LAB2 - >[ 35.97825433  11.89413552  13.05318518]`

2- delta E - CIE 2000:

`print (delta_E(lab1, lab2, method="CIE 2000"))`
`1.73415729242`

Based in this DELTA E, can assume that the two colors are quite similar. Indicating that the spectral distribution of the generated illumiant allow to place the color checker at ANY illuminat, like 4000, 4200.
Do it make any sense, Am I understandi correct?

**Comment by thomasmansencal** [comment, marked as answer]
Hello @xeggbee,

Yes you are correct! Once the reflectances of a surface are obtain, you can compute their tristimulus values under any illumination.

At first glance, your computations above seem to be correct too!

Cheers,

Thomas
